src/Microsoft.Azure.WebJobs.Host/Bindings/BindingProviders/ExactTypeBindingProvider.cs

```diff
@@ -96,7 +102,14 @@ private class ExactBinding : BindingBase<TAttribute>
 
                 IValueProvider vp = new ConstantValueProvider(obj, typeof(TUserType), invokeString);
                 return vp;
-            }            
+            }
+
+            protected override Task<IValueProvider> BuildAsync(TAttribute attrResolved, BindingContext context)
+            {
+                _preBindHook?.Invoke(attrResolved, context);
```

Comment:
Ah, I see what you mean by the "rule" delegate - I think you're right - if I just change the signature there that would work.